File: samples/hello.js
/**
 * iM — Quick Look 预览工具
 *
 * 支持的语法高亮语言列表。
 * 基于 highlight.js — 代码块通过 fenced code block 标记语言，
 * 由 PreviewHTML.makeHTML() 渲染为 GitHub 风格高亮。
 */

const SUPPORTED_LANGUAGES = [
  // 文档
  { id: "markdown",     exts: [".md", ".markdown", ".mdown", ".mkd"] },

  // 配置
  { id: "yaml",         exts: [".yaml", ".yml"] },
  { id: "json",         exts: [".json"],          maxBytes: 2 * 1024 * 1024 },
  { id: "toml",         exts: [".toml"] },
  { id: "xml",          exts: [".xml", ".plist"] },

  // 脚本
  { id: "bash",         exts: [".sh", ".bash", ".zsh"] },

  // 编程语言
  { id: "swift",        exts: [".swift"] },
  { id: "python",       exts: [".py"] },
  { id: "javascript",   exts: [".js"] },

  // 样式
  { id: "css",          exts: [".css"] },
];

// 按扩展名查找
function findByExt(ext) {
  return SUPPORTED_LANGUAGES.find(
    (lang) => lang.exts.includes(ext.toLowerCase())
  );
}

console.table(
  SUPPORTED_LANGUAGES.map((l) => ({
    语言: l.id,
    扩展名: l.exts.join(", "),
    限制: l.maxBytes ? `${l.maxBytes / 1024 / 1024}MB` : "—",
  }))
);

File: samples/hello.py
"""Quick Look 预览文件分类器。

根据文件扩展名将文件路由到对应的预览渲染器。
"""

import json
from pathlib import Path


EXTENSION_MAP: dict[str, str] = {
    ".md": "markdown",
    ".markdown": "markdown",
    ".yaml": "yaml",
    ".yml": "yaml",
    ".json": "json",
    ".toml": "toml",
    ".xml": "xml",
    ".plist": "xml",
    ".sh": "bash",
    ".bash": "bash",
    ".zsh": "bash",
    ".swift": "swift",
    ".py": "python",
    ".js": "javascript",
    ".css": "css",
}


def classify(filepath: str) -> str | None:
    """返回文件对应的 highlight.js 语言标识符。"""
    ext = Path(filepath).suffix.lower()
    return EXTENSION_MAP.get(ext)


def is_large_json(filepath: str, limit_mb: float = 2.0) -> bool:
    """检查 JSON 文件是否超过大小限制。"""
    path = Path(filepath)
    if path.suffix.lower() != ".json":
        return False
    return path.stat().st_size > limit_mb * 1024 * 1024


if __name__ == "__main__":
    for f in ["README.md", "config.json", "script.py", "style.css"]:
        lang = classify(f)
        print(f"{f:>16}  →  {lang}")

    large = is_large_json("huge.json")
    print(f"\nhuge.json 超限: {large}")
